### ISSUE TITLE
Add course materials access to Student Activity Dashboard

### ISSUE THREAD

**Issue by bmcutler** [Submitty/Submitty#7205]
**What problem are you trying to solve with Submitty**
Now that we have data on most recent access per user per item in course materials, we can add this as a column & option to the student activity dashboard.

@cjreed121 Question: If an instructor creates, but then replaces an item on the student activity dashboard (e.g., they update the syllabus), do we delete the record of accesses to the deleted items?  This might be undesirable.  If the student checked the syllabus last week, but then the instructor made an insignificant change to the syllabus (corrected a spelling typo), to display that the student never accessed course materials would be misleading.

**Comment by cjreed121** [comment]
Yes all accesses would be deleted. This is enforced by the code and also by database as their is a foreign key to ensure that course material ID exists. In terms of the student activity dashboard, I'm guessing you would only be able to show their latest access which would display time and material. If 1 item gets replaced I doubt it would incredibly effect this data as courses that do use course materials seem to post quite a lot of content.

A solution to this issue could be allowing for the actual file to be replaced for that particular ID.

**Comment by immortalcodes** [comment]
@bmcutler where do we have this information about recent access per user per item ?

**Comment by cjreed121** [comment]
@immortalcodes I added this feature so I'll explain. Information about who accessed a course material is stored in https://github.com/Submitty/Submitty/blob/main/site/app/entities/course/CourseMaterialAccess.php which is related to https://github.com/Submitty/Submitty/blob/main/site/app/entities/course/CourseMaterial.php. These are doctrine entities and if you are unfamiliar with them I would recommend looking through https://github.com/Submitty/Submitty/blob/main/site/app/controllers/course/CourseMaterialsController.php to see how to perform CRUD operations on them.
